### PR TITLE
Upgrade to IDL 2 and use enum shapes

### PR DESCRIPTION
**Pull request by mullermp**
Also fixes the 50+ warnings we have.

Upgrades to IDL 2

### CODE REVIEW

File: codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/EnumGenerator.java [hunk 1]
```diff
@@ -15,26 +15,26 @@
...
 @SmithyInternalApi
 public final class EnumGenerator extends RubyGeneratorBase {
 
-    private final Shape shape;
+    private final Set<EnumShape> enumShapes;
 
     public EnumGenerator(GenerateEnumDirective<GenerationContext, RubySettings> directive) {
         super(directive);
-        this.shape = directive.shape();
+        this.enumShapes = directive.model().getEnumShapes();
```
Comment (alextwoods) [inline codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/EnumGenerator.java:37]: This still seems wrong - This should be doing directive.shape() and should only process one enum from the directive I'm pretty sure

Comment (mullermp) [reply codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/EnumGenerator.java]: AH that was it. Thanks

**Review by AndrewFossAWS — APPROVED** [review]

File: codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/EnumGenerator.java [hunk 2]
```diff
@@ -15,26 +15,24 @@
...
 @SmithyInternalApi
 public final class EnumGenerator extends RubyGeneratorBase {
 
-    private final Shape shape;
+    private final EnumShape shape;
 
     public EnumGenerator(GenerateEnumDirective<GenerationContext, RubySettings> directive) {
         super(directive);
-        this.shape = directive.shape();
+        this.shape = (EnumShape) directive.shape();
```
Comment (alextwoods) [inline codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/EnumGenerator.java:35]: Use expectEnumShape instead here

**Review by alextwoods — APPROVED** [review]
Nice, looks good overall!

File: codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/EnumGenerator.java (hunk 2 above)
Comment (mullermp) [reply codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/EnumGenerator.java:35]: I had that originally but I copied what int enum shape did. I can change both

**Pull request merged**